frontend/src/metabase/visualizations/lib/apply_axis.js

```diff
@@ -138,14 +139,31 @@ export function applyChartTimeseriesXAxis(
       }
     }
 
-    chart.xAxis().tickFormat(timestamp => {
-      // timestamp is a plain Date object which discards the timezone,
-      // so add it back in so it's formatted correctly
-      const timestampFixed = moment(timestamp)
-        .utcOffset(dataOffset)
-        .format();
-      return formatValue(timestampFixed, {
-        ...chart.settings.column(dimensionColumn),
+    chart.xAxis().tickFormat(rawTimestamp => {
+      // `rawTimestamp` is a plain Date object which discards the timezone. If
+      // we're going to display a time, we need to add it back in so it's
+      // formatted correctly.
+      //
+      // An example:
+      // If we're passed "2019-08-15T03:00:00+03:00", but the data offset is -3
```

Comment:
perhaps a dumb question, but in what situations would the data offset and the timezone attached to a data point differ?

Is this only used if the Reporting Timezone and the User's browser timezone differ?

Reply:
We extract `dataOffset` from the first row returned form the server, so that's how we get reporting timezone. In this function though, d3 is passing us timestamps to format ticks in the user's local time.

> Is this only used if the Reporting Timezone and the User's browser timezone differ?

Yup! If those match, overwriting the offset has no effect.